File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/mappers/elementcache/ApiElementCachePreviewMapper.java
package com.gigigo.orchextra.core.data.api.mappers.elementcache;

import com.gigigo.ggglib.mappers.ExternalClassToModelMapper;
import com.gigigo.orchextra.core.data.api.dto.elementcache.ApiElementCachePreview;
import com.gigigo.orchextra.core.domain.entities.elementcache.ElementCacheBehaviour;
import com.gigigo.orchextra.core.domain.entities.elementcache.ElementCachePreview;

public class ApiElementCachePreviewMapper
  implements ExternalClassToModelMapper<ApiElementCachePreview, ElementCachePreview> {

  @Override public ElementCachePreview externalClassToModel(ApiElementCachePreview data) {
    ElementCachePreview model = new ElementCachePreview();

    model.setText(data.getText());
    model.setImageUrl(data.getImageUrl());
    model.setImageThumb(data.getImageThumb());
    model.setBehaviour(ElementCacheBehaviour.convertStringToEnum(data.getBehaviour()));

    return model;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/mappers/elementcache/ApiElementCachePreviewMapper.kt
package com.gigigo.orchextra.core.data.api.mappers.elementcache

import android.util.Log
import com.gigigo.ggglib.mappers.ExternalClassToModelMapper
import com.gigigo.orchextra.core.data.api.dto.elementcache.ApiElementCachePreview
import com.gigigo.orchextra.core.domain.entities.elementcache.ElementCacheBehaviour
import com.gigigo.orchextra.core.domain.entities.elementcache.ElementCachePreview

class ApiElementCachePreviewMapper : ExternalClassToModelMapper<ApiElementCachePreview, ElementCachePreview> {

  override fun externalClassToModel(data: ApiElementCachePreview): ElementCachePreview {
    val time = System.currentTimeMillis()

    val model = ElementCachePreview()

    with(model) {
      text = data.text
      imageUrl = data.imageUrl
      imageThumb = data.imageThumb
      behaviour = ElementCacheBehaviour.convertStringToEnum(data.behaviour)
    }

    val currentTime = System.currentTimeMillis() - time
    Log.v("TT - ApiElemCachePreview", ("" + currentTime/1000))

    return model
  }
}
